medmodels/medrecord/_overview.py

```diff
@@ -1,109 +1,133 @@
+from medmodels.medrecord.types import (
+    AttributeInfo,
+    Attributes,
+    EdgeIndex,
+    Group,
+    MedRecordAttribute,
+    NodeIndex,
+    NumericAttributeInfo,
+    StringAttributeInfo,
+    TemporalAttributeInfo,
+)
 
 
 def extract_attribute_summary(
-    attribute_dict: Union[Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]],
+    attribute_dictionary: Union[
+        Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]
+    ],
     schema: Optional[AttributesSchema] = None,
-    decimal: int = 2,
-) -> pl.DataFrame:
+) -> Dict[
+    MedRecordAttribute,
+    Union[TemporalAttributeInfo, NumericAttributeInfo, StringAttributeInfo],
+]:
     """Extracts a summary from a node or edge attribute dictionary.
 
-
-    Example:
-    ┌────────────────┬──────────────────────────┐
-    │ Attribute      ┆ Info                     │
-    │ ---            ┆ ---                      │
-    │ str            ┆ str                      │
-    ╞════════════════╪══════════════════════════╡
-    │ diagnosis_time ┆ min: 1962-10-21 00:00:00 │
-    │ diagnosis_time ┆ max: 2024-04-12 00:00:00 │
-    │ duration_days  ┆ min: 0.0                 │
-    │ duration_days  ┆ max: 3416.0              │
-    │ duration_days  ┆ mean: 405.02             │
-    └────────────────┴──────────────────────────┘
-
-
     Args:
         attribute_dict (Union[Dict[EdgeIndex, Attributes], Dict[NodeIndex, Attributes]]):
             Edges or Nodes and their attributes and values.
         schema (Optional[AttributesSchema], optional): Attribute Schema for the group
             nodes or edges. Defaults to None.
-        decimal (int): Decimal points to round the numeric values. Defaults to 2.
+        decimal (int): Decimal points to round the numeric values to. Defaults to 2.
 
     Returns:
-        pl.DataFrame: Summary of node or edge attributes.
+        Dict[MedRecordAttribute, Union[TemporalAttributeInfo, NumericAttributeInfo,
+            StringAttributeInfo]: Summary of node or edge attributes.
     """
-    data = pl.DataFrame(data=[{"id": k, **v} for k, v in attribute_dict.items()])
+    data = pl.DataFrame(data=[{"id": k, **v} for k, v in attribute_dictionary.items()])
 
-    data_dict = {
-        "Attribute": [],
-        "Info": [],
-    }
-
-    attributes = [col for col in data.columns if col != "id"]
-    attributes.sort()
+    data_dict = {}
 
-    if not attributes:
-        return pl.DataFrame({"Attribute": ["-"], "Info": ["-"]})
+    attributes = sorted([col for col in data.columns if col != "id"])
 
     for attribute in attributes:
         attribute_values = data[attribute].drop_nulls()
 
         if len(attribute_values) == 0:
-            attribute_info = ["-"]
-        elif schema and attribute in schema:
+            # to comply with the schema for the attribute info
+            attribute_info = {"values": "-"}
+        elif schema and attribute in schema and schema[attribute][1]:
```

Comment:
could still use an explanation. Line 55, not the line above where you added the comment

Reply:
the `schema[attribute][1]` part is not verbose enough